### PR TITLE
build: add kokoro & ci configs to 1.106.1-patch branch

### PR DESCRIPTION
**Pull request by danielduhh**
This adds update kokoro and ci configs to the `1.106.1-patch` branch in prep for a patch release.

Required for: #844

### CODE REVIEW

**Comment by danielduhh** [comment]
https://github.com/googleapis/java-storage/pull/862 should remove the linkage-monitor check @BenWhitehead

**Comment by danielduhh** [comment]
This is ready to go @BenWhitehead @frankyn. I've added the Kokoro integration & passed the tests.

**Review by frankyn — APPROVED** [review]

File: .kokoro/build.bat [hunk 1]
```diff
@@ -1,3 +1,3 @@
 :: See documentation in type-shell-output.bat
 
-"C:\Program Files\Git\bin\bash.exe" github/java-storage/.kokoro/build.sh
+"C:\Program Files\Git\bin\bash.exe" %~dp0build.sh
```
Comment (frankyn) [inline .kokoro/build.bat:3]: Is this an expected changed?

File: .kokoro/continuous/readme.cfg [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2020 Google LLC
```
Comment (frankyn) [inline .kokoro/continuous/readme.cfg:1]: New files should have 2021 year.

**Review by frankyn — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 3]
```diff
@@ -233,6 +233,17 @@
         </plugin>
       </plugins>
     </pluginManagement>
+    <plugins>
+      <plugin>
+        <groupId>com.coveo</groupId>
+        <artifactId>fmt-maven-plugin</artifactId>
```
Comment (frankyn) [inline pom.xml:239]: Missed this: You don't need to include this plugin into pom.xml.

Comment (danielduhh) [reply pom.xml:239]: Ben added this in https://github.com/googleapis/java-storage/pull/847/commits/64597019d7e3d69688e3309405221c97fa30806c to fix a `UnsupportedClassVersionError` issue

File: .kokoro/build.bat (hunk 1 above)
Comment (danielduhh) [reply .kokoro/build.bat:3]: The kokoro changes are a result of the old branch (mar 2020) being out of date with the latest config

File: .kokoro/continuous/readme.cfg (hunk 2 above)
Comment (danielduhh) [reply .kokoro/continuous/readme.cfg:1]: This is just a copy of what the Kokoro config file looks like now

**Review by frankyn — APPROVED** [review]
The changes are super subtle. Thanks for addressing my concerns, LGTM

**Pull request merged**